### DISCUSSION TITLE
YAML_CPP_INCLUDE_DIR

### DISCUSSION THREAD

**Discussion by quasiquote1** [sparcians/map#328] [Q&A] [answered]
I updated to the latest sparta and started getting this error when running cmake on sparta:

CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
YAML_CPP_INCLUDE_DIR

I was able to fix this by modifying cmake/sparta-config.cmake:

+get_property(YAML_CPP_INCLUDE_DIR TARGET yaml-cpp PROPERTY INTERFACE_INCLUDE_DIRECTORIES)

I am not sure why this error is popping up - I'm using cmake version 3.16.3

Is there a fix for this issue?  Thanks in advance

**Comment by klingaard** [comment]
Which OS is this?

**Reply by quasiquote1** [reply]
Ubuntu 20.04.2

**Reply by klingaard** [reply]
I tested with ubuntu, MacOS and RH8 without issue, but some other folks had the same problem.  You wouldn't know which version of yaml-cpp you have?  I noticed in the envs I tested we have yaml-cpp 0.7.0.

**Reply by quasiquote1** [reply]
I'm using version 0.6.2.  Are you recommending to update to 0.7.0?

**Reply by klingaard** [reply]
If you could try that, I'd appreciate it.  If that solves your issue, I think I'll look into adding the `get_property` line you suggested and make sure it works with both versions.

**Reply by quasiquote1** [reply]
Looks like my version of Ubuntu doesn't support this version.  I'll see if I can find a newer Ubuntu image

**Reply by jackjchen** [reply]
If you invest just a bit of time to use a conda environment, you should get a good env to work with and working combination (0.6.3 yaml-cpp with cmake 3.23.1) on any Ubuntu. The instructions here should work: https://github.com/sparcians/map/tree/master/helios/pipeViewer#prerequisites

**Reply by robbellphd** [reply]
Knute, Jack, not understanding how you already have yaml-cpp0.7. Which ubuntu version did you test? The ubuntu docs say it is in Jammy which will be released on 4/21/2022, 5 days from now. It looks like even the prior Indris has yaml-cpp0.6. Did you download it separately from somewhere? I am also on 20.04 focal, I build cmake 3.23.1, and I still get a cmake error with incompatibility of calling package name "Cppcheck" and the name passed, ie "CPPCHECK."

**Reply by klingaard** [reply]
I started a branch `knutel/yaml_include_issue_328` that has @quasiquote1's fix in it.  Rob, if you want to try that branch, I'd appreciate it.  I'm trying the fix as well.

**Reply by klingaard** [reply]
I'm also using the conda environment that Jack mentions, which alleviate a lot of annoyances.

**Reply by robbellphd** [reply]
@klingaard @quasiquote1 @jackjchen Thanks, the change in that patch allowed the build to work. All the regress and example core tests pass. Using ubuntu 20.04, yaml-cpp 0.6.2, cmake 3.23.1 (manually downloaded from github Kitware and applied). Interesting note, the SchedulerTimeout_test seg faulted on the first regress run. When I reran the full regress, it passed.

**Reply by robbellphd** [reply]
I see that Jammy will be at only cmake 3.22.1 when it is released, don't know if that is enough to work on Sparta out of the box when I update to Jammy. My current Focal was at cmake 3.16.3.

**Reply by klingaard** [reply]
Thanks for trying it out, Rob!  I tested it with RHEL8 with no issues as well.  I'll start a PR.  Thanks, @quasiquote1 for the suggestion!

**Comment by klingaard** [comment]
BTW, to see exactly what Sparta is looking for:
```
cmake -DCMAKE_BUILD_TYPE=Release .. -DSparta_VERBOSE=ON
```

**Reply by jackjchen** [reply]
@robbellphd, I am running ubuntu 20.04 with a conda env which currently will solve to use yaml-cpp 0.6.3 (where I don't see this yaml include problem). The other cmake issue you mention may be just a warning that the optional cppcheck tool is not found. If this is the sole issue, you should be able to continue with building and running in my experience, or just install cppcheck.

**Reply by robbellphd** [reply]
Thanks @jackjchen will look into using conda eventually.

**Comment by ghost** [comment, marked as answer]
Fixed in master via PR #330
